server/datastore/mysql/migrations/tables/20230721161508_QueriesDataMigrator.go

```diff
@@ -0,0 +1,467 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20230721161508, Down_20230721161508)
+}
+
+// This is meant to future-proof this migration, this type is based on the fleet.Query type.
+type _20230719152138_Query struct {
+	TeamID                  *uint
+	TeamIDChar              string
+	ObserverCanRun          bool
+	ScheduleInterval        uint
+	Platform                string
+	MinOsqueryVersion       string
+	AutomationsEnabled      bool
+	LoggingType             string
+	Name                    string
+	Description             string
+	Query                   string
+	Saved                   bool
+	AuthorID                *uint
+	ScheduledQID            uint
+	ScheduledQueryInterval  uint
+	ScheduledQSnapshot      *bool
+	ScheduledQRemoved       *bool
+	ScheduledQueryPlatform  string
+	ScheduledQueryVersion   string
+	ScheduledQueryTimestamp time.Time
+	PackType                string
+	TeamRoles               string
+}
+
+func _20230719152138_QueryName(q _20230719152138_Query) string {
+	return fmt.Sprintf("%s - %d - %s", q.Name, q.ScheduledQID, q.ScheduledQueryTimestamp.Format("Jan _2 15:04:05.000"))
```

Comment:
Ah yeah, mysql only stores timestamps with second precision by default so the decimals for ms will always be 000, but I think it's fine to leave it like that as the mysql precision could change someday and can be changed explicitly, and if ms are available then they'll be part of the name. 

Thinking out loud, for `fleetctl apply` new queries via a yaml file, will that become too annoying to have query names with the timestamp in it? Was that explicitly requested by product to be "name - id"? Just wondering if trying to prevent a relatively unlikely or rare issue could end up being too much trouble/annoyance for everyday use.

Reply:
Product didn't request a specific name format, but we need to handle conflicts somehow, because having two or more schedules targeting the same query is a somewhat common scenario I think, for example:

![image](https://github.com/fleetdm/fleet/assets/8041318/e7c38362-a488-490c-843a-3c06c1b7e899)

Re `fleetctl apply` the thing is that deriving a good descriptive 'query' name that avoids collisions is kind of tricky to do, because the identity of `scheduled_queries` entities are not based on its attributes, i.e. to me it would make a lot of sense to name the query something like `Get USD device type (every 12 hours)` for example, but we can't do that because you can have more than one `scheduled_queries` with the same frequency, etc.

Reply:
Gotcha, yeah that works for me then. :+1: